src/module.cpp

```diff
@@ -460,7 +460,8 @@ instruction_ref module::add_parameter(std::string name, shape s)
 
 instruction_ref module::add_return(std::vector<instruction_ref> args)
 {
-    impl->push_back({builtin::returns{}, {}, std::move(args)});
+    shape instr_shape = args.empty() ? shape{} : args[0]->get_shape();
```

Comment:
This logic should be in the `compute_shape` method. And then it can just do `shape instr_shape = compute_shape(builtin::returns{}, args)`. This is similar to how `insert_instructions` work.

Reply:
@pfultz2 thanks. Makes sense. Should this logic be added to src/include/migraphx/op/reduce.hpp? (There isn't a file yet like that).

Reply:
> Should this logic be added to src/include/migraphx/op/reduce.hpp?

What logic are you referring to?

Reply:
The compute_shape logic for builtin::returns.